### ISSUE TITLE
bug(mat-option): JAWS 2023 does not announce a checked option as selected in a multiple selection mat-select

### ISSUE THREAD

**Issue by walberglez** [angular/components#28214]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

The screen reader JAWS version 2023 does not announce a checked option as selected in a multiple selection mat-select. It does announce unchecked options as selected.

I tried other Windows screen readers, NVDA does not announce it either but Narrator does. So it could be a screen reader implementation issue.

Also, this issue has been discussed before: https://github.com/angular/components/issues/21949 and it was considered to be working as expected. But I want to bring this up again because I consider that the mat-option component is not following the WAI ARIA Listbox pattern completely, please see Note 1 in this section https://www.w3.org/WAI/ARIA/apg/patterns/listbox/#roles_states_properties.

> 1. Some factors to consider when choosing whether to indicate selection with aria-selected or aria-checked are:
Some design systems use aria-selected for single-select widgets and aria-checked for multi-select widgets. In the absence of factors that would make an alternative convention more appropriate, this is a recommended convention.

Mat Option uses aria-select for both single and multiple selection. Based on this recommendation, it should use aria-selected for single selection and aria-checked for multiple selection.

### Reproduction

Steps to reproduce:
1. Start JAWS in a Windows machine.
2. Go to https://material.angular.io/components/select/overview#select-multiple
3. Select the multiple selection dropdown.
4. Navigate through the options using the arrow keys.
5. Select a few options.
6. Navigate through selected and unselected options and check how the screen reader announces them.

### Expected Behavior

I expect screen readers to announce if an option is selected or unselected.

### Actual Behavior

JAWS only announces the status of an unselected option.

### Environment

- Angular: 15.2.9
- CDK/Material: 15.2.9
- Browser(s): Chrome 119.0.6045.199
- Operating System: Windows 10 Enterprise

**Comment by zarend** [comment]
Hello,

Thank you for reporting this. JAWS is one of our supported screen readers.

> JAWS only announces the status of an unselected option.

This just seems to be how JAWS works.

I see that WAI ARIA guide gives aria-checked as an option. That's a good guide, but I don't know of any accessibility requirement that says screen readers need to announces both selected and unselected options.

I don't see enough of a case for changing it to aria-checked at this time :). Currently, we are looking to make a11y improvements by fixing WCAG violations and removing unnecessary live regions.

Do you have a WCAG criteria or some other criteria to support this? OR do you have an issue report from screen reader user or user that relies on ASsistive Technology? That could help convince us on this.

Best Regards,

Zach

**Comment by walberglez** [comment]
Hi Zach, thank you for your quick response.

The application I am working on just went through a 508 conformance testing session and this issue was reported as not in compliance with the WCAG 2.0 SC 1.3.1: https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html

I hope it helps.

Thank you.

**Comment by jeffreinhart** [comment]
Considering this from a perspective of just trying to interact with this component as a visually impaired user, the functionality is simply lacking, WCAG or not.

To understand what an NVDA user would experience, try this:

1. Go to https://material.angular.io/components/select/overview.
2. Activate NVDA.
3. Tab to the Toppings input under H3 for Multiple selection.
4. Press ENTER on the input to expand.
5. TAB to the through the list and press ENTER on list items.

The only announcements I get are the value and list position (e.g. “extra cheese 1 of 6”) and “selected” when pressing ENTER on an item that is not selected. As I tab through the list, I don’t know what is or is not selected and when I press ENTER on an item that is selected, I get no notice that anything happened.

Behavior that would be useful and is not currently present:

- When a list item is focused via tabbing, announce “selected” or “unselected” along with the value and list position.
- When a list item is unselected via ENTER, announce “unselected”.
